### DISCUSSION TITLE
These two codes can not be scanned

### DISCUSSION THREAD

**Discussion by marcelr** [journeyapps/zxing-android-embedded#745] [General]
Hi,

I have these two codes:

![grafik](https://github.com/journeyapps/zxing-android-embedded/assets/956536/b3b73df3-88f3-443c-a8d9-2fe762ea9181)

* On native Android 13 the bottom code is scanned, but not the top one
* Using zxing in my App can't scan both

opts.setDesiredBarcodeFormats(ScanOptions.ALL_CODE_TYPES); // does not help

Any idea how to proceed?

Thank you
Marcel

**Comment by acaliaro** [comment]
I can scan both using Scandit Demo App. The first is a datamatrix. The second a qrcode

**Comment by rkistner** [comment]
These examples are white-on-black instead of the typical black-on-white. To scan them, use `INVERTED_SCAN` or `MIXED_SCAN` mode:

https://github.com/journeyapps/zxing-android-embedded/blob/d09b7c76c3124fbfbd096a65d60b1997f37ff90f/sample/src/main/java/example/zxing/MainActivity.java#L52-L56

**Comment by marcelr** [comment]
Thank you, this does now everything I need:

options.addExtra(Intents.Scan.SCAN_TYPE, Intents.Scan.MIXED_SCAN); // for white on black codes

I didn't notice that the performance of scanning was decreased by allowing mixed scans, probably having this as default is a good idea?
